local/server.go

```diff
@@ -387,7 +387,7 @@ func (a *apiServer) handleAcceptJob(w http.ResponseWriter, r *http.Request, jobI
 		`BUILDKITE_AGENT_META_DATA_QUEUE`:     `default`,
 		`BUILDKITE_REBUILT_FROM_BUILD_ID`:     ``,
 		`BUILDKITE_REBUILT_FROM_BUILD_NUMBER`: ``,
-		`BUILDKITE_PIPELINE_DEFAULT_BRANCH`:   `master`,
+		`BUILDKITE_PIPELINE_DEFAULT_BRANCH`:   getenv("BUILDKITE_CLI_PIPELINE_DEFAULT_BRANCH", "main"),
```

Comment:
Nice idea!

I think given the special context of bulidkite-cli it makes sense to inherit this value from the outer environment. So instead of a new env, perhaps we could use the same env:

```suggestion
		`BUILDKITE_PIPELINE_DEFAULT_BRANCH`:   getenv("BUILDKITE_PIPELINE_DEFAULT_BRANCH", "main"),
```